src/mixins/measurements/base.ts

```diff
@@ -167,33 +96,38 @@ export class BaseMeasurementManager extends BaseDOMListenreeManagerWtihThreeGrou
             }
         }
 
+        // On leave atom, reset the previously intersected atom
         if (!intersects.length) {
-            if (this.highlightedAtom) {
+            if (this.intersectedAtom) {
                 this.setDefaultHexForAtom();
             }
         }
+    };
+
+    createMeasurementLabel(text: string, name: string, position: THREE.Vector3) {
+        const labelManager = new BaseLabelsManager(this.waveStructureGroup, this.waveCamera);
```

Comment:
can be set on the class level and reused for formatting the text, etc.

in coordinates, use CoordinateLabelsManager to get printing of the coordinates

Reply:
See `add labelManager` above